### PR TITLE
Allow users to use either scan or subsonic. Or both!

### PR DESCRIPTION
**Pull request by mayhem**
This PR eliminates the need to scan your collection before a subsonic match. Now you can do a subsonic match or a file system match... or if you want to live on the edge, do both. YOLO!

Guards are in place to not write playlists that make no sense (e.g. m3u with subsonic_id)

### CODE REVIEW

File: lb_content_resolver/model/recording.py [hunk 1]
```diff
@@ -12,7 +34,8 @@ class Meta:
         database = db
 
     id = AutoField()
-    file_path = TextField(null=False, unique=True)
+    file_id = TextField(null=False, unique=True)
+    file_id_type = FileIdTypeField(null=False)
```
Comment (zas) [inline lb_content_resolver/model/recording.py]: I think we have to create few new indexes, one for uniqueness on both fields, and one for each field:

```diff
diff --git a/lb_content_resolver/model/recording.py b/lb_content_resolver/model/recording.py
index 2512aa7..713c570 100644
--- a/lb_content_resolver/model/recording.py
+++ b/lb_content_resolver/model/recording.py
@@ -32,10 +32,14 @@ class Recording(Model):
 
     class Meta:
         database = db
+        indexes = (
+            # create a unique on (file_id, file_id_type)
+            (('file_id', 'file_id_type'), True),
+        )
 
     id = AutoField()
-    file_id = TextField(null=False, unique=True)
-    file_id_type = FileIdTypeField(null=False)
+    file_id = TextField(null=False, index=True)
+    file_id_type = FileIdTypeField(null=False, index=True)
     mtime = TimestampField(null=False)
 
     artist_name = TextField(null=True)
```

**Review by zas — CHANGES_REQUESTED** [review]
It looks very good, I just have a remark about indexes

File: lb_content_resolver/model/recording.py (hunk 1 above)
Comment (mayhem) [reply lb_content_resolver/model/recording.py]: Good call!

Comment (mayhem) [reply lb_content_resolver/model/recording.py]: Applied.

**Review by zas — APPROVED** [review]
LGTM

**Pull request merged**